sdk/v1/client_test.go

```diff
@@ -261,21 +269,37 @@ func TestClient_PostTasksCount(t *testing.T) {
 				So(task.NumberOfDocuments, ShouldEqual, expectedTask.NumberOfDocuments)
 			})
 
+			Convey("And an ETag is returned", func() {
+				So(respETag, ShouldNotBeNil)
+				So(respETag, ShouldResemble, testETag)
+			})
+
 			Convey("And client.Do should be called once with the expected parameters", func() {
 				doCalls := httpClient.DoCalls()
 				So(doCalls, ShouldHaveLength, 1)
 				So(doCalls[0].Req.URL.Path, ShouldEqual, pathToCheck)
 				So(doCalls[0].Req.Body, ShouldResemble, io.NopCloser(bytes.NewReader(testPayload)))
+				expectedIfMatchHeader := make([]string, 1)
+				expectedIfMatchHeader[0] = "*"
+				So(doCalls[0].Req.Header[ifMatchHeader], ShouldResemble, expectedIfMatchHeader)
 			})
 		})
 	})
 	Convey("Given a 500 response", t, func() {
-		httpClient := newMockHTTPClient(&http.Response{StatusCode: http.StatusInternalServerError}, nil)
+		httpClient := newMockHTTPClient(
+			&http.Response{
+				StatusCode: http.StatusInternalServerError,
+				Body:       nil,
```

Comment:
(minor): you don't have to explicitly specify body and header is nil, by default if you don't specify the body and header attribute it will be nil only i guess.